reference/CloudPay-API-Specification.yaml

```diff
@@ -1381,6 +1381,115 @@ paths:
 
         **Note** *Only one of the four authorisation parameters are required*
     parameters: []
+  /api/v1/account/profile/billing:
+    put:
+      summary: Update the customer's Stripe billing profile
+      operationId: put-api-v1-account-profile-billing
+      parameters:
+        - $ref: '#/components/parameters/apiJwtTokenParam'
+        - $ref: '#/components/parameters/apiSessionIdParam'
+        - $ref: '#/components/parameters/thirdPartyAuthCookie'
+        - $ref: '#/components/parameters/authCookie'
+      tags:
+        - Account
+        - Profile
+        - Billing details
+      responses:
+        '202':
+          description: Accepted
+        '400':
+          description: Bad Request
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '403':
+          description: Forbidden
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+        '500':
+          description: Error Response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/ErrorResponse'
+      description: |-
+        ![In Review](https://img.shields.io/badge/Status-In%20Review-green?style=flat-square)
+        The update customer billing profile endpoint allows a customer to update their billing details.
+        Please be aware that the customer billing profile depends on Provider - Stripe or Paypal.
+        **Note** *Only one of the four authorisation parameters are required*
+      requestBody:
+        content:
+          application/json:
+            schema:
+              description: 'Customer Stripe billing profile'
+              type: object
+              properties:
+                  AddressCountry:
+                    type: string
+                  AddressCity:
+                    type: string
+                  AddressLine1:
+                    type: string
+                  AddressLine2:
+                    type: string
+                  AddressState:
+                    type: string
+                  AddressZip:
+                    type: string
+                  CardDetails:
+                    type: object
+                    items:
+                      properties:
+                        Number:
+                          type: string
+                          pattern: '^\d+$'
```

Comment:
The example does not match the regex pattern here, searching for card number regex gives some results such as

- \b4[0-9]{12}(?:[0-9]{3})?\b
- ^(?:4[0-9]{12}(?:[0-9]{3})?|[25][1-7][0-9]{14}|6(?:011|5[0-9][0-9])[0-9]{12}|3[47][0-9]{13}|3(?:0[0-5]|[68][0-9])[0-9]{11}|(?:2131|1800|35\d{3})\d{11})$

These would invalidate numbers such as '1111222233334444' - so depends if we want to have a strict regex like these or a simpler one such as '<4 numbers><space><4 numbers><space><4 numbers><space><4 numbers>'

Reply:
seems there are a lot of different type of cards, validation should happen in application o for now I will remove pattern

Reply:
Added pattern: '^\d{8,20}$' I have seen this for card number in payments provider api